x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/logging/Throttler.java

```diff
@@ -7,152 +7,222 @@
-     *                       is reached
+     * @param loggingInterval the frequency to run a task to emit repeated log messages
      * @param threadPool a thread pool for running a scheduled task to clear the internal stats
      */
-    public Throttler(TimeValue resetInterval, TimeValue durationToWait, ThreadPool threadPool) {
-        this(resetInterval, durationToWait, Clock.systemUTC(), threadPool, new ConcurrentHashMap<>());
+    public Throttler(TimeValue loggingInterval, ThreadPool threadPool) {
+        this(loggingInterval, Clock.systemUTC(), threadPool, new ConcurrentHashMap<>());
+    }
+
+    /**
+     * @param oldThrottler a previous throttler that is being replaced
+     * @param loggingInterval the frequency to run a task to emit repeated log messages
+     */
+    public Throttler(Throttler oldThrottler, TimeValue loggingInterval) {
+        this(loggingInterval, oldThrottler.clock, oldThrottler.threadPool, new ConcurrentHashMap<>(oldThrottler.logExecutors));
     }
 
     /**
      * This should only be used directly for testing.
      */
-    Throttler(
-        TimeValue resetInterval,
-        TimeValue durationToWait,
-        Clock clock,
-        ThreadPool threadPool,
-        ConcurrentMap<String, LogExecutor> logExecutors
-    ) {
-        Objects.requireNonNull(durationToWait);
-        Objects.requireNonNull(threadPool);
-
-        this.resetInterval = Objects.requireNonNull(resetInterval);
-        this.durationToWait = Duration.ofMillis(durationToWait.millis());
+    Throttler(TimeValue loggingInterval, Clock clock, ThreadPool threadPool, ConcurrentMap<String, LogExecutor> logExecutors) {
+        this.threadPool = Objects.requireNonNull(threadPool);
+        this.loggingInterval = Objects.requireNonNull(loggingInterval);
         this.clock = Objects.requireNonNull(clock);
         this.logExecutors = Objects.requireNonNull(logExecutors);
+    }
 
-        this.cancellableTask.set(startResetTask(threadPool));
+    public void init() {
+        cancellableTask.set(startRepeatingLogEmitter());
     }
 
-    private Scheduler.Cancellable startResetTask(ThreadPool threadPool) {
-        classLogger.debug(() -> format("Reset task scheduled with interval [%s]", resetInterval));
+    private Scheduler.Cancellable startRepeatingLogEmitter() {
+        classLogger.debug(() -> Strings.format("Scheduling repeating log emitter with interval [%s]", loggingInterval));
+
+        return threadPool.scheduleWithFixedDelay(this::emitRepeatedLogs, loggingInterval, threadPool.executor(UTILITY_THREAD_POOL_NAME));
+    }
+
+    private void emitRepeatedLogs() {
+        if (isRunning.get() == false) {
+            return;
+        }
+
+        final ReentrantReadWriteLock.WriteLock writeLock = lock.writeLock();
 
-        return threadPool.scheduleWithFixedDelay(logExecutors::clear, resetInterval, threadPool.executor(UTILITY_THREAD_POOL_NAME));
+        writeLock.lock();
```

Comment:
Locks so that we don't get inconsistencies while the logging the repeated messages. Otherwise the counts could be incorrect.